pkg/nvml/init.go

```diff
@@ -32,15 +35,10 @@ var nvml *dl.DynamicLibrary
 // nvml.Init()
 func Init() Return {
 	lib := dl.New(nvmlLibraryName, nvmlLibraryLoadFlags)
-	if lib == nil {
-		panic(fmt.Sprintf("error instantiating DynamicLibrary for %s", nvmlLibraryName))
-	}
-
 	err := lib.Open()
 	if err != nil {
-		panic(fmt.Sprintf("error opening %s: %v", nvmlLibraryName, err))
+		return ERROR_LIBRARY_NOT_FOUND
```

Comment:
Although this is cleaner than a `panic` from certain perspectives this is not a documented error returned by the underlying `nvmlInit` function.

What we tend to do internally is use a wrapper (e.g. `tryNvmlInit`) that recovers from the panic and converts the `Return` to an `error` instead.

With that said, I'll let @klueska weigh in on this change.

Reply:
This comes up frequently enough as a perceived issue that I'm inclined to go ahead and change it since there is in fact an error called `ERROR_LIBRARY_NOT_FOUND`.